File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/data/network/api/UploadImageApi.java
package com.lattis.ellipse.data.network.api;

import com.lattis.ellipse.data.network.model.response.uploadImage.UploadImageResponse;

import io.reactivex.Observable;
import okhttp3.MultipartBody;
import retrofit2.http.Multipart;
import retrofit2.http.POST;
import retrofit2.http.Part;
import retrofit2.http.Query;

/**
 * Created by Velo Labs Android on 06-04-2017.
 */

public interface UploadImageApi {
    @Multipart
    @POST("misc/upload")
    Observable<UploadImageResponse> uploadImage(@Query("type") String upload_Type,
                                                @Part MultipartBody.Part  filePart);

}

File: velolabs_repos/lattis_android_v2/data/src/main/java/com/lattis/data/repository/datasources/api/uploadimage/UploadImageApi.kt
package com.lattis.data.repository.datasources.api.uploadimage

import com.lattis.data.entity.response.uploadimage.UploadImageResponse
import io.reactivex.rxjava3.core.Observable
import okhttp3.MultipartBody
import retrofit2.http.Multipart
import retrofit2.http.POST
import retrofit2.http.Part
import retrofit2.http.Query

interface UploadImageApi {
    @Multipart
    @POST("api/misc/upload")
    fun uploadImage(
        @Query("type") upload_Type: String,
        @Part filePart: MultipartBody.Part
    ): Observable<UploadImageResponse>
}